help.py

```diff
@@ -41,3 +42,276 @@ def datetime_to_epoch_millis(date: datetime) -> int:
+    handler = logging.FileHandler(os.path.join(logs_dir, log_file_name))
+    handler.setLevel(logging.DEBUG)
+    handler.setFormatter(logging.Formatter(log_format))
+    root.addHandler(handler)
+    
+    return log
+
+def check_product_args_structure(edr: str, product_args: dict) -> dict:
+    if product_args == {}:
+        return {"result": True, "edr": edr, "required_fields": "No arguments were supplied"}
+
+    product_check = [k for k, v in product_args.items()]
+    
+    if edr == "cbc":
+        required_fields = ['device_group', 'device_policy']
+    elif edr == "cbr":
+        required_fields = ['sensor_group']
+    elif edr == "s1":
+        required_fields = ['deep_visibility']
+    else:
+        return {"result": False, "edr": edr, "required_fields": required_fields}
+
+    for field in required_fields:
+        if field not in product_check:
+            return {"result": False, "edr": edr, "required_fields": required_fields}
+
+    return {"result": True, "edr": edr, "required_fields": required_fields}
+
+
+def check_credentials_structure(edr, creds: dict) -> dict:
+    # Check credential provided for the bare minimum arguments for execution.
+    # This function checks for the presence of needed arguments; it does not check for valid input.
+    if edr in ["cbc", "cbr"]:
+        return {"result": True, "edr": edr, "required_fields": "Assuming credentials file is stored in required location"}
+    
+    cred_check = [k for k, v in creds.items() if v]
+
+    if 'creds_file' and 'profile' in cred_check:
+        return {"result": True, "edr": edr, "required_fields": ["creds_file", "profile"]}
+
+    if edr == "cbc":
+        required_fields = ['url', 'token', 'org_key']
+    elif edr == "cbr":
+        required_fields = ['url', 'token']
+    elif edr == "cortex":
+        required_fields = ['api_key', 'url', 'api_key_id', 'auth_type']
+    elif edr == "dfe":
+        if 'token' in cred_check:
+            return {"result": True, "edr": edr, "required_fields": "token"}
+        else:
+            required_fields = ['tenantId', 'appId', 'appSecret']
+    elif edr == "s1":
+        state = False
+        required_fields = ['site_ids', 'account_ids', 'account_names']
+        for i in required_fields:
+            if i in cred_check:
+                state = True
+        if state:
+            required_fields = ['url', 'token']
+            for i in required_fields:
+                if i in cred_check:
+                    state = True
+            
+        return {"result": state, "edr": edr, "required_fields": "'url', 'token', and one of 'site_ids', 'account_ids', or 'account_names'"}
+        
+    else:
+        return {"result": False, "edr": edr, "required_fields": required_fields}
+
+    for field in required_fields:
+        if field not in cred_check:
+            return {"result": False, "edr": edr, "required_fields": required_fields}
+
+    return {"result": True, "edr": edr, "required_fields": required_fields}
```

Comment:
Rather than have multiple `return` statements, the only thing that really changes is the `result` key. We could have something like below and then just have a final return at the end of the function
```python
# (pseudo-code)
# by default, it fails
state = False
if cbc and passing:
  state = True
elif cbr and passing:
  state = True
...
return {"result": state, "edr": edr, "required_fields": required_fields}
```